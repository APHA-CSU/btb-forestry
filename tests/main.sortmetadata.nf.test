nextflow_process {

    name "Test Process sortmetadata"
    script "main.nf"
    process "SORT_META_DATA"

    test("Should sort metadata") {

        when {
            params {
                today = "01Jan25"
            }
            process {
                """
                input[0] = file("${projectDir}/tests/data/metaTest.csv")
                input[1] = file("${projectDir}/tests/data/moveTest.csv")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
