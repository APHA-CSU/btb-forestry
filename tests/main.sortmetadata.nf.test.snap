{
    "Should sort metadata": {
        "content": [
            {
                "0": [
                    "sortedMetadata_01Jan25.csv:md5,5ce00c3cfdc4c285665a8d2929f3ba13"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-22T12:59:00.57844051"
    }
}